dtc.c

```diff
@@ -327,14 +327,14 @@ int main(int argc, char *argv[])
 	process_checks(force, dti);
 
 	if (auto_label_aliases)
-		generate_label_tree(dti, "aliases", false);
+		generate_label_tree(dti, xstrdup("aliases"), false);
```

Comment:
However, for the `generate_*_tree()` functions I think we should do better.  For these *every* caller is passing a static string, so we should do the `xstrdup()` inside the functions rather than in the callers.

Reply:
If they are all static, why do you want to dup the string? Also for the consistency of the API, it would be best to pick one style.

Reply:
Because those strings get inserted as node names into a section of the tree, and the node names are supposed to be allocated and "owned" by the node structure.   I see your point about API, but I wasn't really thinking of these `generate` functions as part of the same API as the lower level API to the live tree - they're more complex operations. 

However, given the fact that it's demonstrably likely for people to get this wrong, I think it's better to make `name_node()` and the rest also do the strdup.  It comes at the cost of an additional copy of the lexer tokens, but really that's a trivial cost for a safer API.

I'm right now doing final tests on some patches to make that change.